ui/src/components/common/RequestCustodianRelationship.tsx

```diff
@@ -69,11 +75,16 @@ const RequestCustodianRelationship: React.FC<Props> = ({ role }) => {
             }}
         >
             <Form.Group className='inline-form-group'>
-                <Form.Input
-                    label='Custodian'
-                    placeholder='Enter ID'
-                    value={custodianId}
-                    onChange={e => setCustodianId(e.currentTarget.value)}/>
+                <ContractSelect
+                    className='custodian-select-container'
+                    allowAdditions
```

Comment:
You probably don't need to `allowAdditions` here, because all custodians are publicly registered so there's no need to add a private custodian ID. 

You should also know that whenever making use of `allowAdditions`, you need to use the `setAddition` prop to get the added input and use it. (This is because the existing options are keyed by contractId, but added options are just user input, so they get keyed by the value itself). 

The select inside `Exchange/InviteParticipant.tsx` has an example of an addition-enabled dropdown